[docs] Update README.md to remove outdated information (#34490)
expo/expo: docs/README.md

This is the public documentation for **Expo**, its SDK, client, and services (**EAS**). This documentation is built using Next.js and you can access it online at https://docs.expo.dev/.

> > [!NOTE] 

Reviewer: Whitespace issue.

> **Contributors:** Please make sure that you edit the docs in the **pages/versions/unversioned** for SDK reference if you want your changes to apply to the next SDK version too!

> > [!NOTE]

Reviewer: ```suggestion
> [!TIP]
```

Let's use an another callout theme to avoid stacking two same types below each other, other options:
* https://github.com/orgs/community/discussions/16925

This is the public documentation for **Expo**, its SDK, client, and services (**EAS**). This documentation is built using Next.js and you can access it online at https://docs.expo.dev/.

> > [!NOTE] **Contributors:** Please make sure that you edit the docs in the **pages/versions/unversioned** for SDK reference if you want your changes to apply to the next SDK version too!

Author: ```suggestion
> [!NOTE] 
> **Contributors:** Please make sure that you edit the docs in the **pages/versions/unversioned** for SDK reference if you want your changes to apply to the next SDK version too!
```

The documentation is divided into four main sections:

- **Home**: Provides a guided path from starting a project from scratch to deploying it to app stores.
- **Guides**: General purpose and fundamental guides that help you understand how Expo works and how to use it.

> - **EAS**: DEtailed documentation for all EAS services.

Reviewer: ```suggestion
- **EAS**: Detailed documentation for all EAS services.
```

Typo.

### Prose linter

> When you are done writing or editing docs, run the following script to lint your docs for style and grammar based on [Expo's writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md):

Reviewer: ```suggestion
When you are done writing or editing docs, run the following script to lint your docs for style and grammar based on [Expo's writing style guide](/guides/Expo%20Documentation%20Writing%20Style%20Guide.md):
```

Nit, we can shorten the link.

```sh
yarn run lint-prose
```

We use [Vale](https://vale.sh/) to lint our docs.

#### Switch off Prose linter

For exceptional cases, you can switch off the prose linter for a specific line or block of text by adding by using a [comment delimiter](https://vale.sh/docs/keys/commentdelimiters):

```mdx
{/* vale off */}

> This is some text that will not be linted by Vale.

Author: ```suggestion
This is some text that will be ignored by Vale.
```

- Open [http://localhost:3002/](http://localhost:3002/) in the browser and go to the API doc to see the changes you have made. Make sure to select the right SDK version to see the changes in the left sidebar.

<details>
<summary>Updating versioned documentation data after SDK lifecycle is latest or already released</summary>

When you need to update versioned documentation data late in the SDK lifecycle, follow these steps:

1. Ensure the related code change exists on the `main` branch and `sdk-*` branch.

2. Make branch changes:

> - Switch to the sdk-\* branch
> - Run command: `et gdad -p expo-library --sdk 52`, where `expo-library` is the library you want to update and `52` is the SDK version.
> - Store/shelf the changes locally (you can use `git stash`)
> 
> 3. Update main branch:
> 
> - Switch back to main branch

Reviewer: ```suggestion
- Switch to the `sdk-\*` branch
- Run command: `et gdad -p expo-library --sdk 52`, where `expo-library` is the library you want to update and `52` is the SDK version.
- Store/shelf the changes locally (you can use `git stash`)

3. Update `main` branch:

- Switch back to `main` branch
```

Formatting nits.

| ----------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `videoId`   | **Required**. The ID of the video from YouTube. You can find this in the URL of the video. For example, in `https://www.youtube.com/watch?v=Gk7RHDWsLsQ`, the ID is `Gk7RHDWsLsQ`. |
| title       | **Required**. The title of the video.                                                                                                                                              |
| description | **Optional**. The description of the video.                                                                                                                                        |

### Add code block

Code blocks are a great way to add code snippets to our docs. We leverage the usual code block Markdown syntax, but it's expanded to support code block titles and additional params.

<!-- prettier-ignore -->

> ````mdx

Reviewer: This change does not look correct.
